packages/backend-common/src/compat/legacy/legacy.ts

```diff
@@ -96,30 +166,52 @@ export function makeLegacyPlugin<
         env.registerInit({
           deps: {
             ...envMapping,
-            _router: coreServices.httpRouter,
-            _auth: coreServices.auth,
+            $$router: coreServices.httpRouter,
+            $$auth: coreServices.auth,
+            $$userInfo: coreServices.userInfo,
+            $$config: coreServices.rootConfig,
+            $$logger: coreServices.logger,
           },
-          async init({ _router, _auth, ...envDeps }) {
+          async init({
+            $$auth,
+            $$config,
+            $$logger,
+            $$router,
+            $$userInfo,
+            ...envDeps
+          }) {
             const { default: createRouter } = await createRouterImport;
             const pluginEnv = Object.fromEntries(
               Object.entries(envDeps).map(([key, dep]) => {
                 const transform = envTransforms[key];
                 if (transform) {
                   return [key, transform(dep)];
                 }
-                if (key === 'tokenManager') {
-                  return [
-                    key,
-                    wrapTokenManager(dep as TokenManager, _auth as AuthService),
-                  ];
-                }
                 return [key, dep];
               }),
             );
-            const router = await createRouter(
-              pluginEnv as TransformedEnv<TEnv, TEnvTransforms>,
+
+            const auth = $$auth as typeof coreServices.auth.T;
```

Comment:
wait why are these casts needed?

Reply:
Because of the mapped object type these end up being the service `| ServiceRef<TEnv[key]>`